app/views/AdminView/components/Appendices/index.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {
+  useState, useEffect, useMemo, useCallback,
+} from 'react';
+import { CircularProgress } from '@mui/material';
+import { RecordDefaults, AppendixType } from '@/common';
+import AppendixEditor from '@/components/AppendixEditor';
+import sanitizeHtml from 'sanitize-html';
+import api from '@/services/api';
+import snackbar from '@/services/SnackbarUtils';
+import DataTable from '@/components/DataTable';
+import columnDefs from './columnDefs';
+
+type TemplateWithAppendix = RecordDefaults & { appendix?: AppendixType };
+
+function Appendices(): JSX.Element {
+  const [templates, setTemplates] = useState<RecordDefaults[]>([]);
+  const [appendices, setAppendices] = useState<TemplateWithAppendix[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editingData, setEditingData] = useState(null);
+
+  // Grab templates
+  useEffect(() => {
+    const getTemplates = async () => {
+      const templatesResp = await api.get('/templates').request();
+      setTemplates(templatesResp);
+    };
+    getTemplates();
+  }, []);
+
+  // Grab template appendices
+  useEffect(() => {
+    const templateIdToObj = {};
```

Comment:
similarly this hook also needs a cancelled flag

Reply:
[ef72650](https://github.com/bcgsc/pori_ipr_client/pull/181/commits/ef72650885d99aacf47b8fb03b3261fda8fa5e1a)